plugins/utils/ollama.nix

```diff
@@ -0,0 +1,226 @@
+{
+  lib,
+  helpers,
+  config,
+  pkgs,
+  ...
+}:
+with lib; let
+  cfg = config.plugins.ollama;
+
+  actionOptionType = with helpers.nixvimTypes;
+    oneOf [
+      rawLua
+      (enum [
+        "display"
+        "replace"
+        "insert"
+        "display_replace"
+        "display_insert"
+        "display_prompt"
+      ])
+      (submodule {
+        options = {
+          fn = helpers.mkNullOrStrLuaFnOr (enum [false]) ''
+            fun(prompt: table): Ollama.PromptActionResponseCallback
+
+            Example:
+            ```lua
+              function(prompt)
+                -- This function is called when the prompt is selected
+                -- just before sending the prompt to the LLM.
+                -- Useful for setting up UI or other state.
+
+                -- Return a function that will be used as a callback
+                -- when a response is received.
+                ---@type Ollama.PromptActionResponseCallback
+                return function(body, job)
+                  -- body is a table of the json response
+                  -- body.response is the response text received
+
+                  -- job is the plenary.job object when opts.stream = true
+                  -- job is nil otherwise
+                end
+
+              end
+            ```
+          '';
+
+          opts = {
+            stream = helpers.defaultNullOpts.mkBool false ''
+              Whether to stream the response.
+            '';
+          };
+        };
+      })
+    ];
+in {
+  meta.maintainers = [maintainers.GaetanLepage];
+
+  options.plugins.ollama =
+    helpers.extraOptionsOptions
+    // {
+      enable = mkEnableOption "ollama.nvim";
+
+      package = helpers.mkPackageOption "ollama.nvim" pkgs.vimPlugins.ollama-nvim;
+
+      model = helpers.defaultNullOpts.mkStr "mistral" ''
+        The default model to use.
+      '';
+
+      prompts = let
+        promptOptions = {
+          prompt = mkOption {
+            type = types.str;
```

Comment:
maybeRaw types.str, if you want to dynamically create the prompt

Reply:
Indeed !